endpoints/openrtb2/amp_auction.go

```diff
@@ -578,21 +588,46 @@ func setAmpExt(site *openrtb2.Site, value string) {
 	}
 }
 
-func readPolicy(consent string) (privacy.PolicyWriter, error) {
-	if len(consent) == 0 {
+// setRegExtGDPR sets regs.ext.gdpr to either 0 or 1 only if it was set in the request query
+func setRegExtGDPR(gdprApplies *bool, req *openrtb2.BidRequest) error {
```

Comment:
This code should live in the privacy/gdpr package - ideally within the consent writer.

Reply:
If we move to the consent writer, we'll need the `GDPR.Validate(ampParams.Consent)` call to be true and according to the issue #1817 specs, we don't need the consent string to be true in order to set `regs.ext.gdpr`, hence the reason it belongs on this package (I believe). Thoughts?

Reply:
Moved to the `gdpr` package.